src/control/zoom/ZoomControl.cpp

```diff
@@ -321,15 +321,19 @@ void ZoomControl::zoomPresentation() {
 void ZoomControl::setZoomFitMode(bool isZoomFitMode) {
     if (this->zoomFitMode != isZoomFitMode) {
         this->zoomFitMode = isZoomFitMode;
-        this->control->fireActionSelected(GROUP_ZOOM_FIT, isZoomFitMode ? ACTION_ZOOM_FIT : ACTION_NOT_SELECTED);
+        this->control->fireActionSelected(GROUP_ZOOM_FIT, this->zoomFitMode ? ACTION_ZOOM_FIT : ACTION_NOT_SELECTED);
     }
 
-    if (isZoomFitMode) {
+    if (this->isZoomFitMode()) {
         zoomFit();
     }
 }
 
-auto ZoomControl::isZoomFitMode() const -> bool { return this->zoomFitMode; }
+auto ZoomControl::isZoomFitMode() const -> bool {
+    // Todo(fabian): Remove this fix and make both modes possible in parallel, after fixing the infinite loop
+    auto infiniteLoopFixup = !this->zoomPresentationMode && !this->control->getSettings()->isShowPairedPages();
```

Comment:
It would be nice to have a comment that explains why `infiniteLoopFixup` is set this way...